cromwellApiClient/src/main/scala/cromwell/api/CromwellClient.scala

```diff
@@ -155,19 +135,71 @@ object CromwellClient {
     override def getMessage: String = message + ": " + httpResponse.toString
   }
 
-  private[api] def insertSecrets(options: Option[String], refreshToken: Option[String]): Option[String] = {
+  /**
+    * Optionally replace a json value. Returns the original json if:
+    * - The jsonOption is None
+    * - The key is not found in the json
+    * - The valueOption is None
+    *
+    * @param jsonOption The optional json
+    * @param key The key
+    * @param valueOption The optional value
+    * @return The json with the modified value, or the original json
+    */
+  def replaceJson(jsonOption: Option[String], key: String, valueOption: Option[String]): Option[String] = {
+    val newJsonOption = for {
+      value <- valueOption
+      json <- jsonOption
+      newJson = replaceJson(json, key, value)
+    } yield newJson
+
+    newJsonOption orElse jsonOption
+  }
+
+  /**
+    * Replace a json value. Returns the original json if the key is not found in the json.
+    *
+    * @param json The json
+    * @param key The key
+    * @param value The value
+    * @return The json with the modified value, or the original json
+    */
+  def replaceJson(json: String, key: String, value: String): String = {
     import DefaultJsonProtocol._
-    val tokenKey = "refresh_token"
-
-    val secretOptions = for {
-      refreshTokenValue <- refreshToken
-      optionsValue <- options
-      optionsMap = optionsValue.parseJson.asJsObject.convertTo[Map[String, JsValue]]
-      if optionsMap.contains(tokenKey)
-      secretMap = optionsMap.updated(tokenKey, JsString(refreshTokenValue))
-      secretValue = secretMap.toJson.toString
-    } yield secretValue
-
-    secretOptions orElse options
+    val newJsonOption = for {
+      _ <- Option(json)
+      if json.contains(key)
+      map = json.parseJson.asJsObject.convertTo[Map[String, JsValue]]
+      if map.contains(key)
```

Comment:
If one tries to to decode a bad json value, say `{`, the code would blow up on line 172 in the `parseJson`. We only want the code to parse the json iff the string contains the key.

Later on 173, we check if the map contains the key just in case someone has done something weird like `{"these_are_a_few_of_my_favorite_things": ["refresh_token"]}`.